### PR TITLE
[FSSDK-9785] handle duplicate experiment keys with a warning

### PR DESCRIPTION
**Pull request by jaeopt**
## Summary
When duplicate experiment keys are found in a datafile, SDK returns a correct experimentMap in OptimizelyConfig:
1. the experimentMap will contain the experiment later in the datafile experiments list.
2. add a warning log about "Duplicate experiment keys found in datafile: {key-name}"

## Test plan
- add a test for OptimizelyConfig with duplicate experiment keys

## Issues
- [FSSDK-9785](https://jira.sso.episerver.net/browse/FSSDK-9785)

### CODE REVIEW

File: core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigService.java [hunk 1]
```diff
@@ -31,6 +33,8 @@ public class OptimizelyConfigService {
     private Map<String, List<FeatureVariable>> featureIdToVariablesMap = new HashMap<>();
     private Map<String, OptimizelyExperiment> experimentMapByExperimentId = new HashMap<>();
 
+    private static final Logger logger = LoggerFactory.getLogger(OptimizelyConfigService.class);
+
     public OptimizelyConfigService(ProjectConfig projectConfig) {
```
Comment (mikechu-optimizely) [inline core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigService.java:38]: Is it worth adding an overload/optional parameter to the constructor to allow logger to be created from outside?  🤔

File: core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigService.java
Comment (mikechu-optimizely) [inline core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigService.java:1]: I think the copyright year needs to be updated.

File: core-api/src/test/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigServiceTest.java [hunk 2]
```diff
@@ -15,23 +15,29 @@
...
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.mockito.runners.MockitoJUnitRunner;
 
 import java.util.*;
 import static java.util.Arrays.asList;
 import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
```
Comment (mikechu-optimizely) [inline core-api/src/test/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigServiceTest.java:29]: I love the mockitos  🍸

File: core-api/src/test/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigServiceTest.java
Comment (mikechu-optimizely) [inline core-api/src/test/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigServiceTest.java:1]: Copyright year update here too?

File: java-quickstart/src/main/java/com/optimizely/Example.java
Comment (mikechu-optimizely) [inline java-quickstart/src/main/java/com/optimizely/Example.java:1]: Grrr someone needs to develop a pre-commit hook that scans text files for copyright years.

**Review by mikechu-optimizely — APPROVED** [review]
LGTM no showstoppers but some thoughts 👇

File: core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigService.java (hunk 1 above)
Comment (jaeopt) [reply core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigService.java:38]: java-sdk uses SFL4J and creates a logger for each class. It may be worth, but we do not follow the logger injection pattern here.

File: java-quickstart/src/main/java/com/optimizely/Example.java
Comment (jaeopt) [reply java-quickstart/src/main/java/com/optimizely/Example.java:1]: Agreed :)

**Review by mikechu-optimizely — APPROVED** [review]
LGTM
